endpoints/authentication/src/main.cpp

```diff
@@ -46,10 +46,54 @@ namespace beast = boost::beast; // from <boost/beast.hpp>
 namespace net   = boost::asio;  // from <boost/asio.hpp>
 // clang-format on
 
-using body_arguments = std::unordered_map<std::string, std::string>;
-
 namespace irods::http::handler
 {
+	auto create_host_field(boost::urls::url_view _url, std::string_view _port) -> std::string {
+		if ((_port.compare("443") == 0 && _url.scheme_id() == boost::urls::scheme::https) ||
+			(_port.compare("80") == 0 && _url.scheme_id() == boost::urls::scheme::http)) {
+			return _url.host();
+		}
+		return fmt::format("{}:{}", _url.host(), _port);
+	}
+
+	auto remove_client_from_body_if_confidential_client(body_arguments _body) -> body_arguments {
+		const static bool has_client_secret{irods::http::globals::oidc_configuration().contains("client_secret")};
+
+		if (has_client_secret) {
+			_body.erase("client_id");
+		}
+
+		return _body;
+	}
```

Comment:
Still not sure if this is hacky or not.

Reply:
or substitute xxxxxxxxx.    depends on the use case for this function.     we do the same when logging configuration values in other projects / libraries.

Reply:
Is this because of logging?

Alternatively, you could overwrite it with `[REDACTED]` or something.

Reply:
Think we're good on this one, unless we prefer the inverse, i.e. add client_id later compared to removing it?

Reply:
Just circling back to this. What are we trying to solve here? I need for context to understand the issue.

Reply:
This got separated since the replies in the reviews threw me off:

> Not for logging. OAuth spec allows you to auth either in the header (preferred) or the body. Some OpenID Providers may not appreciate finding a spare client_id in the body after finding it in the header.

I think this is the context you're looking for too?

Reply:
I see. And so the approach you've chosen is to leave this as is _(i.e. remove it from the body)_?

Reply:
Yep, not opposed to the other way, though.

Reply:
I don't have strong opinions on it.

Let's leave it as is for now.